### PR TITLE
feat: EcdsaMockAVSDeployer avs-370

### PR DESCRIPTION
**Pull request by Gajesh2007**
https://linear.app/eigenlabs/issue/AVS-370/write-ecdsaavsmockdeployer

### CODE REVIEW

File: test/harnesses/ECDSAStakeRegistryHarness.sol [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import "../../src/unaudited/ECDSAStakeRegistry.sol";
+
+// Wrapper around the ECDSAStakeRegistry contract that exposes the internal functions for unit testing.
+contract ECDSAStakeRegistryHarness is ECDSAStakeRegistry {
+    constructor(
+        IDelegationManager _delegationManager
+    ) ECDSAStakeRegistry(_delegationManager) {
+    }
+
+    function recordOperatorStakeUpdate(address operator, uint96 newStake) external returns (int256) {
```
Comment (stevennevins) [inline test/harnesses/ECDSAStakeRegistryHarness.sol:13]: Feels like this is unnecessary, we have the ability to change the value for the operator on the DelegationManagerMock and you could call `updateOperators` to pull that value and update both the individual operator's weight and total weight

File: test/utils/ECDSAMockAVSDeployer.sol [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+    /**
+     * @notice registers operator with coordinator
+     */
+    function _registerOperatorWithCoordinator(
+        address operator,
+        uint256 quorumBitmap,
+        uint96 stake
+    ) internal {
+        // quorumBitmap can only have 192 least significant bits
+        quorumBitmap &= MAX_QUORUM_BITMAP;
```
Comment (stevennevins) [inline test/utils/ECDSAMockAVSDeployer.sol]: the quorumBitmap stuff can be removed

File: test/utils/ECDSAMockAVSDeployer.sol [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+
+    address ejector = address(uint160(uint256(keccak256("ejector"))));
+
+    address defaultOperator = address(uint160(uint256(keccak256("defaultOperator"))));
+    bytes32 defaultOperatorId;
+    uint96 defaultStake = 1 ether;
+    uint8 defaultQuorumNumber = 0;
+
+    uint32 defaultMaxOperatorCount = 10;
+    uint16 defaultKickBIPsOfOperatorStake = 15_000;
```
Comment (stevennevins) [inline test/utils/ECDSAMockAVSDeployer.sol]: operator kick params can be removed

File: test/utils/ECDSAMockAVSDeployer.sol [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+            operatorMetadatas[i].operatorId = keccak256(abi.encodePacked("operatorId", pseudoRandomNumber, i));
+            operatorMetadatas[i].stakes = new uint96[](maxQuorumsToRegisterFor);
+            for (uint256 j = 0; j < maxQuorumsToRegisterFor; j++) {
+                operatorMetadatas[i].stakes[j] = uint96(
+                    uint64(uint256(keccak256(abi.encodePacked("stakes", pseudoRandomNumber, i, j))))
+                );
+            }
+        }
+
+        // get the index in quorumBitmaps of each operator in each quorum in the order they will register
```
Comment (stevennevins) [inline test/utils/ECDSAMockAVSDeployer.sol:288]: pretty sure most of this can be removed

File: test/utils/ECDSAMockAVSDeployer.sol [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+
+    function _incrementAddress(address start, uint256 inc) internal pure returns (address) {
+        return address(uint160(uint256(uint160(start) + inc)));
+    }
+
+    function _incrementBytes32(bytes32 start, uint256 inc) internal pure returns (bytes32) {
+        return bytes32(uint256(start) + inc);
+    }
+
+    function _signOperatorChurnApproval(
```
Comment (stevennevins) [inline test/utils/ECDSAMockAVSDeployer.sol:366]: can be removed

File: test/utils/ECDSAMockAVSDeployer.sol (hunk 2 above)
Comment (Gajesh2007) [reply test/utils/ECDSAMockAVSDeployer.sol]: why

Comment (stevennevins) [reply test/utils/ECDSAMockAVSDeployer.sol]: it's not used in the ECDSA contracts

Comment (Gajesh2007) [reply test/utils/ECDSAMockAVSDeployer.sol]: oh dang. yes.

File: test/utils/ECDSAMockAVSDeployer.sol [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+
+import {Slasher} from "eigenlayer-contracts/src/contracts/core/Slasher.sol";
+import {ISlasher} from "eigenlayer-contracts/src/contracts/interfaces/ISlasher.sol";
+import {PauserRegistry} from "eigenlayer-contracts/src/contracts/permissions/PauserRegistry.sol";
+import {IStrategy} from "eigenlayer-contracts/src/contracts/interfaces/IStrategy.sol";
+import {ISignatureUtils} from "eigenlayer-contracts/src/contracts/interfaces/ISignatureUtils.sol";
+import {BitmapUtils} from "../../src/libraries/BitmapUtils.sol";
```
Comment (stevennevins) [inline test/utils/ECDSAMockAVSDeployer.sol:12]: can be removed

File: test/utils/ECDSAMockAVSDeployer.sol [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+
+import {OperatorStateRetriever} from "../../src/OperatorStateRetriever.sol";
```
Comment (stevennevins) [inline test/utils/ECDSAMockAVSDeployer.sol:14]: can be removed

File: test/utils/ECDSAMockAVSDeployer.sol [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+    bytes32 defaultSalt = bytes32(uint256(keccak256("defaultSalt")));
+
+    address ejector = address(uint160(uint256(keccak256("ejector"))));
+
+    address defaultOperator = address(uint160(uint256(keccak256("defaultOperator"))));
+    bytes32 defaultOperatorId;
+    uint96 defaultStake = 1 ether;
+    uint8 defaultQuorumNumber = 0;
+
+    uint8 numQuorums = 192;
```
Comment (stevennevins) [inline test/utils/ECDSAMockAVSDeployer.sol:86]: can remove

File: test/utils/ECDSAMockAVSDeployer.sol [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+
+    StrategyParams[] strategies;
+    IStrategy strategy;
+
+    uint8 maxQuorumsToRegisterFor = 4;
```
Comment (stevennevins) [inline test/utils/ECDSAMockAVSDeployer.sol:91]: can remove

File: test/utils/ECDSAMockAVSDeployer.sol (hunk 4 above)
Comment (stevennevins) [reply test/utils/ECDSAMockAVSDeployer.sol:288]: pretty sure most of this can be removed

**Pull request closed without merging**